### PR TITLE
onboarding clevertap

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/219)

### CODE REVIEW

File: integrations/Clevertap/browser.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    window.clevertap.onUserLogin.push({
+      Site: payload,
+    });
+  }
+
+  track(rudderElement) {
+    logger.debug("in clevertap track");
+    const { event, properties } = rudderElement.message;
+    if (properties) {
+      window.clevertap.event.push(event, properties);
```
[the remark this answers is not in the document]
Comment (utsabc) [reply integrations/Clevertap/browser.js:136]: this is only supported in charged events we should use a validator for other events for checking if arrays or objects are are present

File: integrations/Clevertap/browser.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  isReady() {
+    logger.debug("in clevertap isReady");
+    return !!window.clevertap && window.clevertap.logout !== "undefined";
+  }
+
+  identify(rudderElement) {
+    logger.debug("in clevertap identify");
+
+    const { message } = rudderElement;
+    if (!(message.context && message.context.traits)) {
```
[the remark this answers is not in the document]
Comment (utsabc) [reply integrations/Clevertap/browser.js:81]: Wehn identify is called without traits or id i.e rudderanalytics.identify() by default, SDK is generating an empty object for traits.
Screenshot 2021-03-12 at 5 31 16 PM

File: integrations/Clevertap/browser.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    window.clevertap.enablePersonalization = true;
+    window.clevertap.account.push({ id: this.accountId });
+    if (this.region && this.region !== "none") {
+      window.clevertap.region.push(this.region);
+    }
+  }
+
+  isLoaded() {
+    logger.debug("in clevertap isLoaded");
+    return !!window.clevertap && window.clevertap.logout !== "undefined";
```
[the remark this answers is not in the document]
Comment (utsabc) [reply integrations/Clevertap/browser.js:69]: This is fixed

File: integrations/Clevertap/browser.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  init() {
+    logger.debug("===in init Clevertap===");
+    const sourceUrl =
+      document.location.protocol == "https:"
+        ? "https://d2r1yp2w7bby2u.cloudfront.net/js/a.js"
+        : "http://static.clevertap.com/js/a.js";
+
+    ScriptLoader("clevertap-integration", sourceUrl);
```
[the remark this answers is not in the document]
Comment (utsabc) [reply integrations/Clevertap/browser.js:51]: done

File: integrations/Clevertap/browser.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (utsabc) [reply integrations/Clevertap/browser.js:136]: Added validator

File: utils/utils.js [hunk 5]
```diff
@@ -592,6 +592,20 @@ function getDefinedTraits(message) {
   return traitsValue;
 }
 
+/**
+ * To check if a variable is storing object or not
+ */
+const isObject = (obj) => {
+  return Object.prototype.toString.call(obj) === '[object Object]';
```
Comment (sayan-mitra) [inline utils/utils.js:599]: let's use the **type** method present in the utils

File: utils/utils.js [hunk 6]
```diff
@@ -592,6 +592,20 @@ function getDefinedTraits(message) {
...
+};
+
+/**
+ * To check if a variable is storing object or not
+ */
+const isArray = (obj) => {
+  return Object.prototype.toString.call(obj) === '[object Array]';
```
Comment (sayan-mitra) [inline utils/utils.js:606]: same as above

**Pull request merged**